### DISCUSSION TITLE
What behavior when data loss ?

### DISCUSSION THREAD

**Discussion by terrierscript** [planetscale/discussion#227] [FAQs] [answered]
I append deploy request like those query

before
```
CREATE TABLE foo (
  `id` varchar(191) NOT NULL,
  `name` varchar(191)
)
```

after
```
CREATE TABLE foo (
  `id` varchar(191) NOT NULL,
  `name` varchar(191) NOT NULL
)
```

This deploy request warning about data loss, and I force deployed.
After deploy this change, `Schema` tab show deployed table (name is NOT NULL) but actual data schema is not changed.

What happen when this request deployed?

**Comment by jgreet** [comment, marked as answer]
Sounds like something could have gone wrong there. Maybe you can [open a ticket](https://support.planetscale.com/hc/en-us/requests/new) with a link to deploy request in question?

What I would expect to happen is that the deploy request errors. Given the table and a couple of rows:

```sql
nn/|⚠ main ⚠|> describe foo;
+-------+--------------+------+-----+---------+-------+
| Field | Type         | Null | Key | Default | Extra |
+-------+--------------+------+-----+---------+-------+
| id    | varchar(191) | NO   | PRI | NULL    |       |
| name  | varchar(191) | YES  |     | NULL    |       |
+-------+--------------+------+-----+---------+-------+
nn/|⚠ main ⚠|> select * from foo;
+----+------+
| id | name |
+----+------+
| 1  | bar  |
| 2  | NULL |
+----+------+
```

When you attempt to deploy this change:

```sql
nn/test> alter table foo modify name varchar(191) not null;
```

You should see something like this:

<img width="600" alt="image" src="https://user-images.githubusercontent.com/391331/177659168-ed1b67f9-02c3-4a09-8522-e1849ee4e701.png">

Which you could fix by cleaning up the data in main, or adding a default value for `name`.

```sql
nn/test> alter table foo modify name varchar(191) not null default "some-name";
```
